drivers/wdm/audio/hdaudbus/pdo.cpp

```diff
@@ -78,18 +79,22 @@ HDA_PDOQueryId(
 
     if (IoStack->Parameters.QueryId.IdType == BusQueryInstanceID)
     {
-        UNIMPLEMENTED;
-
-        // FIXME
-        swprintf(DeviceName, L"%08x", 1);
-        Length = wcslen(DeviceName) + 20;
+        Status = RtlStringCbPrintfW(DeviceName,
+                                    sizeof(DeviceName),
+                                    L"%02x%02x",
+                                    DeviceExtension->Codec->Addr,
+                                    DeviceExtension->AudioGroup->NodeId);
+        NT_ASSERT(NT_SUCCESS(Status));
+        Length = wcslen(DeviceName) + 1;
 
         /* allocate result buffer*/
         Device = (LPWSTR)AllocateItem(PagedPool, Length * sizeof(WCHAR));
         if (!Device)
             return STATUS_INSUFFICIENT_RESOURCES;
 
-        swprintf(Device, L"%08x", 1);
+        NT_VERIFY(NT_SUCCESS(RtlStringCbCopyW(Device,
```

Comment:
Not a fan of function calls inside (imbricated) macros; why not using the Status variable as you do just above?